backend/src/test/java/com/speakfluid/backend/entities/DialogueTests.java

```diff
@@ -1,19 +1,59 @@
 package com.speakfluid.backend.entities;
 
+import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.Test;
 
+import java.util.ArrayList;
+import java.util.Arrays;
+import static org.junit.jupiter.api.Assertions.*;
+
+
 /**
  * Tests for the Dialogue class.
+ * Since behaviour of generic class works the same for any given generic type, Dialogue<WozMessage> is tested as a representative.
  *
- * @author  Kai Zhuang
- * @version 1.0
- * @since   2022-11-19
+ * @author  Kai Zhuang, Zoey Zhang
+ * @version 2.0
+ * @since   2022-11-26
  */
 
 class DialogueTests {
+    static Dialogue<WozMessage> d1;
+    static ArrayList<WozMessage> botMsg1;
+    static ArrayList<WozMessage> userMsg1;
+    static WozMessage m1 = new com.speakfluid.backend.entities.WozMessage("text", "Hi, here are the locations of our stores");
```

Comment:
Wondering why it's `com.speakfluid.backend.entities.WozMessage` instead of just `WozMessage`? You imported all the entities on the top do you should be able to make it just WozMessage

Reply:
Also, the trace_type for these WozMessages should be 'request' and 'response'

Reply:
I will change the trace type, the test for Dialogue still needs some changes and I will let you know when the final version is up!!! Thank you so so much for the throughout review!!!!!!

Reply:
It's updated!